src/dbworker.cpp

```diff
@@ -107,6 +109,13 @@ void DBWorker::init()
             QSqlQuery query = prepare(db_schema[i]);
             execute(query);
         }
+    } else {
+        // Limit history size to 2000 entries
+        QSqlQuery cleanupHistory = prepare(QString("DELETE FROM browser_history WHERE id NOT IN (SELECT id from "\
+                                                   "browser_history ORDER BY date DESC LIMIT %1);").arg(MAX_BROWSER_HISTORY_SIZE));
```

Comment:
micro-optimization: a better way to extrapolate a string with a constant is to use the STR macro. This way the string would be extrapolated in build time, not run time.